### DISCUSSION TITLE
Content-type application/json vs application/x-www-form-urlencoded

### DISCUSSION THREAD

**Discussion by welyab** [aws/graph-explorer#308] [Q&A]
Hello everyone!!

I received a a neptune instance running on aws cloud where I am able to access via kubectl port-forward. Querying database on Postman works fine using application/x-www-form-urlencoded. Here a inform query parameters and possible arguments to the query.

But Graph Explorer is requesting neptune with application/json, and database responses with bad request (the same if I request application/json in Postman). It is possible configure Graph Explorer do make request using application/x-www-form-urlencoded?

**Comment by kmcginnes** [comment]
Hi @welyab, thanks for reaching out.

We have been discussing your issue internally. The scenario, as you described it, should be supported by Neptune.

Perhaps something besides the `Content-Type` is causing the `400 - Bad Request` response. I'm curious if you can perform a basic query using curl with JSON.

```bash
curl -X POST -d '{"gremlin":"g.V().limit(1)"}' https://your-neptune-endpoint:port/gremlin
```

https://docs.aws.amazon.com/neptune/latest/userguide/access-graph-gremlin-rest.html#access-graph-gremlin-rest-connect

**Reply by welyab4syngenta** [reply]
Hey @kmcginnes. Thank you for response.
Our application was constructed using OpenCypher. Graph Explorer makes application/json requests that our Neptune don't support (I do not know why yet).

But I just try your example using Grremlin language and Neptune responds well. I will try to use Graph Explorer with gremilin soon.
